src/compiler.ts

```diff
@@ -9836,6 +9849,13 @@ export class Compiler extends DiagnosticEmitter {
         return this.checkFeatureEnabled(Feature.REFERENCE_TYPES, reportNode)
             && this.checkFeatureEnabled(Feature.GC, reportNode);
       }
+      case TypeKind.STRINGREF:
+      case TypeKind.STRINGVIEW_WTF8:
+      case TypeKind.STRINGVIEW_WTF16:
+      case TypeKind.STRINGVIEW_ITER: {
+        return this.checkFeatureEnabled(Feature.REFERENCE_TYPES, reportNode)
+            && this.checkFeatureEnabled(Feature.STRINGREF, reportNode);
+      }
```

Comment:
Some stringref ops required GC feature. like `string.new_wtf8_array` / `string.new_wtf16_array`. Will be great add TODO about this and more precisely checks when these ops will be avalable